Foundation/FileHandle.swift

```diff
@@ -211,7 +211,46 @@ open class FileHandle : NSObject {
             fatalError("\(error)")
         }
     }
-    
+
+    private func _fstat() throws -> stat {
+        var statbuf = stat()
+        if fstat(_fd, &statbuf) < 0 {
+            throw _NSErrorWithErrno(errno, reading: true)
+        }
+        return statbuf
+    }
+
+    private func _fileSize() throws -> Int64 {
+        let statbuf = try _fstat()
+        return Int64(statbuf.st_size)
+    }
+
+    private var _isRegularFile: Bool? = nil
+    private var _blockSize: Int? = nil
```

Comment:
This is unlikely to change unless the volume is changed.  Do we need to consider any side effects from operations like `chroot` altering the volume while the application is executing?